app/split_integer.py

```diff
@@ -1,7 +1,10 @@
 def split_integer(value: int, number_of_parts: int) -> list:
-    parts = []
-    for parts_left in range(number_of_parts, 0, -1):
-        next_number = value // parts_left
-        parts.append(value // parts_left)
-        value -= next_number
-    return parts
+    if number_of_parts == 0:
+        raise ValueError("number_of_parts cannot be zero")
+    if value < number_of_parts:
+        raise ValueError("value cannot be less than number_of_parts")
+
+    base = value // number_of_parts
+    remainder = value % number_of_parts
+    res = [base + 1 if i < remainder else base for i in range(number_of_parts)]
+    return sorted(res)
```

Comment:
The function sorts the result before returning it. If the order of the parts is not important, this is fine. However, if the order matters, you might want to return the list as it is without sorting.